src/util/evented.js

```diff
@@ -82,7 +81,7 @@ class Evented {
             }
 
             if (this._eventedParent) {
-                this._eventedParent.fire(type, util.extend({}, data, this._eventedParentData));
+                this._eventedParent.fire(type, util.extend({}, data, typeof this._eventedParentData === 'function' ? this._eventedParentData() : this._eventedParentData));
```

Comment:
Could you add a test that ensures the `eventedParentData` function is called per emit and passes on the latest & greatest data?